dataverse-webapp/src/main/webapp/roles-edit.xhtml

```diff
@@ -68,9 +82,11 @@
             <div id="editRoleButtonPanel" class="button-block">
                 <p:commandLink type="button" styleClass="btn btn-default" 
                                value="#{bundle.saveChanges}" 
-                               oncomplete="PF('roleForm').hide()"
-                               update="roleDisplay availableRoles @([id$=Messages])"
-                               actionListener="#{managePermissionsPage.updateRole}"/>
+                               oncomplete="if (args &amp;&amp; !args.validationFailed) PF('roleForm').hide();"
+                               update="roleDisplay availableRoles editRolePanel @([id$=Messages])"
+                               actionListener="#{managePermissionsPage.updateRole}">
+                    <f:param name="DO_VALIDATION" value="true"/>
```

Comment:
Why we add conditional validation?

Reply:
I've copy-pasted this approach from roles-assign. What is done here is we turn validation ON only when we try to submit a form that compromises validations. Without it we have failed validation messages on form load (when all the fields are empty thus not validate)

Reply:
Are you sure that this is the case? Because condition is only in required params. And then `f:validateLenght` should still be active

Reply:
Since `required` and `f:validatelength minimumlength="1"` are basically the same thing there was no issue with conditional validation. It would be a problem if we wanted to check for minimal field length greater than 1. However as mentioned below, after a talk with Wojtek, we are happy to just check for `!empty` so `validatelength` can be removed.